rfcs/000-search-revamp.md

```diff
@@ -0,0 +1,124 @@
+- Feature Name: Search Revamp
+- Start Date: 2021-09-20
+- RFC PR: todo
+
+# Search Revamp
+
+## Summary
+
+Revamp search with a cleaner backend implementation, more robust queries, better customizability, and
+making use of modern v7 ES features.  
+
+## Motivation
+
+At Lyft, search enhancement are the #1 most frequent Amundsen feedback. This is no surprise, as the current 
+technical state of search has not changed much from MVP.
+
+We see search as serving two distinct (but related) use cases:
+  - known-data searches, which have a starting point (e.g. tables with a `user_id` column in the `production` schema) 
+    and are more interested in listing the available data matching a known condition
+  - discovery searches, which are looking to find previously-unknown data that would be relevant for solving
+    one's problem
+
+Although we believe Amundsen has been particularly lacking in the discovery search category, our proposal is seeking
+to improve the experience of both search types. 
+
+
+## UI/UX Details
+
+Although the bulk of proposed changes will be to search result quality and relevance, we're also proposing
+a handful of UI/UX changes to filters. We're currently working on designs and will share those as soon as they're
```

Comment:
One idea that came up multiple times in our user sessions is better feedback on aggregated search results.

For example if I am looking for `transactions` data and Amundsen returns 1902 results it's still confusing for some users and they'd like to know top 5 schemas from which those results are coming to get `bird's eye view` on the query made by user. It's just one example but introducing aggregates into search revamp could be also part of UX/UI change.

Reply:
Very solid direction. Would be a huge improvement, and we hear the same feedback around our parts. At the moment I'm not sure we can fit this into the scope of our proposed enhancements, but let me run this by a few folks. At the very least, we can plan ahead and think about how to add this down the road so we don't make suboptimal choices now.